text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,573 @@
+package are not satisfied by the selected target.
+
+This feature should be used when a package clearly does not support all targets. For example:
+`io-uring` requires `cfg(target_os = "linux")`, `gloo` requires `cfg(target_family = "wasm")`, and
+`riscv` requires `cfg(any(target_arch = "riscv32", target_arch = "riscv64"))`.
+
+This feature increases cargo's knowledge of a package. For example, when working in a workspace
+where some packages are for a platform with `target_os = "none"`, and some others are tools that
+require a desktop OS, using `supported-targets` makes `cargo <command>` ignore packages which have
+`supported-targets` that are not satisfied by the selected target.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The `supported-targets` field is an optional key that tells cargo which targets the package can be
+built for.
+```toml
+[package]
+# ...
+supported-targets = 'cfg(any(target_os = "linux", target_os = "macos"))'
+```
+The value of this field must respect the [`cfg` syntax](https://doc.rust-lang.org/reference/conditional-compilation.html),
+and does __not__ accept `cfg(test)`, `cfg(debug_assertions)`, nor `cfg(proc_macro)` as configuration options.
+A malformed `supported-targets` field will raise an error.
+
+If the `supported-targets` field is not present, then the package is assumed to support all targets. That is,
+the default value is `'cfg(all())'` (understood as `cfg(true)`).
+
+When a `cargo` build command (e.g. `check`, `build`, `run`, `clippy`) is run, it checks that the
+selected target satisfies the `supported-targets` of the package being built. If it does not, the
+package is skipped or an error is raised, depending on how [`cargo` was invoked](ignoring-builds).
+However, `supported-targets` is _only_ checked for commands that take a `--target` option and does
+not affect other commands (e.g., `cargo fmt`).
+
+As this field is limited to local development, `cargo package` / `cargo publish` will strip it from `Cargo.toml`.
+Including the field in the `.crate` file is left as a [future possibility](#future-possibilities) for now.
+
+This field is subject to [workspace inheritance](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table).
+
+## Ignoring builds for unsupported targets
+[ignoring-builds]: #igonring-builds-for-unsupported-targets
+
+If cargo is invoked in a workspace or virtual workspace without specifying a package as
+build-target, then `cargo` skips any package that does not support the selected target. If a package
+is specified using `--package` or if `cargo` is invoked on a single package, and the selected target
+does not satisfy the `supported-targets` of the package, then an error is raised.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Once implemented this does not yet solve the problem of dependency pruning in `Cargo.lock`, which
+  is a common use case for this feature.
```

Comment:
To me, it is misleading to state this hear and not list it as a high likelihood future possibility.  In trying to come up with a rewording, I was leaning towards just removing this
```suggestion
```